### ISSUE TITLE
Website2018

### ISSUE THREAD

**Issue by findoslice** [compsoc-edinburgh/website2017#94]
The time has come for us to bid farewell to our much beloved website2017, out with the old in with the new etc.

For those who don't know, CompSoc has a tradition of building a new website every year and this year shouldn't be an exception! The main things we need to consider are:
- [ ] Which framework should be used?
For the past couple of years we've used jekyll (a static website generator), however we should still consider all options.

- [ ] Do we want to use any boilerplates (like skeleton css or bootstrap)?

- [ ] Decide on a rough layout/site map
Are there pages we should have that weren't in website2017? Are there any we should remove?

- [ ] Assign foundation tasks to people
While anybody can contribute to the website, it's much easier to get people to get involved once we have the basic foundations set up (i.e. a stylesheet, skeleton html templates etc.)

This list is in no way exhaustive and if you have any input for any of the tasks, want to get involved, or think I've missed something out that should be considered - please don't hesitate to say in the discussion :)

**Comment by neanias** [comment]
Just a note to say that building a new website every year wasn’t supposed to be a tradition. This year was supposed to be the final major overhaul to make it easier to update year on year...

**Comment by findoslice** [comment]
@neanias ok issue has been updated to reflect that, I do think we should still consider areas for improvement though

**Comment by neanias** [comment]
Oh, of course, please do scope out improvements, but try to not immediately jump on the rebuild bandwagon

**Comment by findoslice** [comment]
Your right that it is an ongoing thing, however we do also have a bit of a calm period at the moment in which people have a bit more time to commit to larger restructuring and enhancement should we find the need for it. Now this is issue is essentially a place to try and get suggestions I guess.

**Comment by rphi** [comment]
Unpopular opinion incoming but could we potentially rename this repo to simply "Website" and then have each year's iteration listed as a version tag/release as opposed to a new repository each time? Just for the purpose of Gittyness considering it has commit history anyway...

**Comment by findoslice** [comment]
Yeah I agree entirely, I like the idea of having each years iteration as a release as well. Would also be nice if each year's iteration were viewable somewhere - like as a subdomain on the website, like 2017.comp-soc.com, or something similar

**Comment by qaisjp** [comment]
this wouldn't make sense as each project starts off as a bare repo. bare
branches are fairly hacky imo.

if a future techsec decides to rebuild the website, old links won't be
broken.

also, the year isn't to signify the academic year, it's essentially a
version number. we don't know what was v1.0 so we just use the year the
project was created

**Comment by findoslice** [comment]
I think it would be nice to have different "releases" as a way of showing what the website looked like over time, so the evolution is more clear. I think it would just be a nice nostalgic thing.

**Comment by rphi** [comment]
Yes, orphan branches are a little dodgy, but very doable, as for version numbers, we have each year on a new branch, with release names simply being the year. So for branches have like `17-master` and `17-dev`.

Alternative take, if each iteration was its own repository, it would be trivial to give each its own GitHub Pages site, and say 5 years down the line, that's going to be a minimum of 10 branches that we'd have to leave open if we weren't going for separate repositories, and that's before feature branches, where we'd have to be careful to ensure everyone prefixes their branches correctly.
